be/src/storage/rowset/rowset.cpp

```diff
@@ -167,24 +167,33 @@ Status Rowset::init() {
     return Status::OK();
 }
 
+StatusOr<std::shared_ptr<Segment>> Rowset::_load_segment(int32_t idx, const TabletSchemaCSPtr& schema,
+                                                         std::shared_ptr<FileSystem>& fs,
+                                                         const FooterPointerPB* partial_rowset_footer) {
+    size_t footer_size_hint = 16 * 1024;
+    std::string seg_path = segment_file_path(_rowset_path, rowset_id(), idx);
+    FileInfo seg_info{.path = seg_path, .encryption_meta = rowset_meta()->get_segment_encryption_meta(idx)};
+    auto res = Segment::open(fs, seg_info, idx, schema, &footer_size_hint, partial_rowset_footer);
+    if (!res.ok()) {
+        auto st = res.status().clone_and_prepend(fmt::format(
+                "Load segment failed tablet:{} rowset:{} rssid:{} seg:{} path:{}", _rowset_meta->tablet_id(),
+                rowset_id().to_string(), _rowset_meta->get_rowset_seg_id(), idx, seg_path));
+        LOG(WARNING) << st.message();
+        return st;
+    }
+    return res;
+}
+
 // use partial_rowset_footer to indicate the segment footer position and size
 // if partial_rowset_footer is nullptr, the segment_footer is at the end of the segment_file
 Status Rowset::do_load() {
     ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(_rowset_path));
     _segments.clear();
-    size_t footer_size_hint = 16 * 1024;
```

Comment:
`footer_size_hint` may be updated in the first segment load and be used to read footer in the next segment load.